baseapp/baseapp_e2e/rest_framework/views.py

```diff
@@ -31,7 +33,14 @@ def load_data(self, request):
 
     @action(detail=False, methods=["POST"])
     def flush_data(self, request):
-        management.call_command(flush.Command(), verbosity=0, interactive=False)
+        connection = connections[DEFAULT_DB_ALIAS]
+        style = no_style()
+        sql_statements = sql_flush(style, connection)
+
+        # Changing the constraints to IMMEDIATE to avoid issues with circular foreign keys like users<->profiles
+        sql_statements = ["SET CONSTRAINTS ALL IMMEDIATE;"] + sql_statements
```

Comment:
I had to add this `SET CONSTRAINTS ALL IMMEDIATE;` to work around the `pending trigger events` issue when trying to truncate all tables. Even though all tables were listed in the `TRUNCATE` command it was still triggering this issue.